Autopilot/AttitudeManager/Src/MahonyAHRS.cpp

```diff
@@ -0,0 +1,232 @@
+//=====================================================================================================
+// MahonyAHRS.c
+//=====================================================================================================
+//
+// Madgwick's implementation of Mayhony's AHRS algorithm.
+// See: http://www.x-io.co.uk/node/8#open_source_ahrs_and_imu_algorithms
+//
+// Date			Author			Notes
+// 29/09/2011	SOH Madgwick    Initial release
+// 02/10/2011	SOH Madgwick	Optimised for reduced CPU load
+//
+//=====================================================================================================
+
+//---------------------------------------------------------------------------------------------------
+// Header files
+
+#include "MahonyAHRS.h"
+#include <math.h>
+
+//---------------------------------------------------------------------------------------------------
+// Definitions
+
+#define sampleFreq	160.0f			// sample frequency in Hz
+#define twoKpDef	(2.0f * .5f)	// 2 * proportional gain
+#define twoKiDef	(2.0f * 0.1f)	// 2 * integral gain
+
+//---------------------------------------------------------------------------------------------------
+// Variable definitions
+
+volatile float twoKp = twoKpDef;											// 2 * proportional gain (Kp)
+volatile float twoKi = twoKiDef;											// 2 * integral gain (Ki)
+volatile float q0 = 0.52513f, q1 = 0.051342f, q2 = 0.014727f, q3 = 0.84934f;					// quaternion of sensor frame relative to auxiliary frame
```

Comment:
I'm curious, where do these initial values come from ?

Reply:
An initial rotation I temporarily changed to match testing data. Setting it back to default now.